test/bonjour.js

```diff
@@ -136,8 +136,10 @@ test('bonjour.find - all services', function (bonjour, t) {
         t.equal(found[1], 'Foo Bar')
         t.equal(found[2], 'Invalid')
         t.equal(found[3], 'Sub Foo')
-        bonjour.destroy()
-        t.end()
+        setTimeout(function () {
```

Comment:
It's actually down to the same kind of reason as the built-in bonjour test for `bonjour.find()`. We're not actually interested in any of the other services, so we set a small timeout to ensure that the services are all up before bonjour.destroy()ing them (ie. they're all up in the next event tick).

If we didn't, we'd potentially try and kill a socket which hadn't yet been bound, which would cause a cancellation error on the socket (not a massive deal, but would cause the test to break).